### DISCUSSION TITLE
Auth seems to work only by refreshing

### DISCUSSION THREAD

**Discussion by WebDevSachin** [nextauthjs/next-auth#5563] [Help]
### Question 💬

Hi, i am creating a application and using next auth with middleware for protecting the routes. But after login it redirect to the desired page. After that i try to visit any protected route it redirects me to the login page. But if i refresh the page after login then it works fine. I am sure i am missing something. Thanks in advance.

### How to reproduce ☕️

I used middleware and signin with credentials if successful then i used router.push to home page. I tried to use redirect true option but error handling was bit confusing for me.

**Comment by balthazur** [comment]
I have the same problem, trying to router.push() to authenticated page after successfull signin("credentials",..: i found that the next-auth session token is not set in the request. It is available only after refreshing the page.
